ultravox/tools/ds_tool/caching.py

```diff
@@ -42,9 +43,25 @@ def __init__(self, client: tts.Client, provider: str):
         self._client = client
         self._base_path = os.path.join(".cache/ds_tool/tts", provider)
 
-    def tts(self, text: str, voice: Optional[str] = None) -> bytes:
+    @overload
+    def tts(self, text: str, voice: Optional[str] = None) -> bytes: ...
+
+    @overload
+    def tts(self, text: List[str], voice: Optional[str] = None) -> List[bytes]: ...
+
+    def tts(
+        self, text: Union[str, List[str]], voice: Optional[str] = None
+    ) -> Union[bytes, List[bytes]]:
```

Comment:
TTS cache diverged from the original TTS interface. I could've modified the original, but it didn't seem necessary. Also doing it at a layer above/below seemed too complicated since the voices needed to be the same.

Reply:
To verify, do you mean not to "overload" the `tts` function or just let the underlying class handle resolving the voice name?
Asking since the comment is about the latter but the lines highlighted are about the former.

Reply:
Added `resolve_voice`.

Reply:
The main thing I was concerned about was the details of the random voice selection leaking out of the tts impl code, which you've addressed.